code/modules/mob/living/simple_animal/hostile/headcrab.dm

```diff
@@ -196,3 +203,223 @@
+			aggro_vision_range = gonarch_aggro_vision_range
+			vision_range = gonarch_standard_vision_range
+			speed = gonarch_standard_speed
+		if(homesick)
+			homesick = 0
+			step_towards(src, home_nest_turf, speed)
+		make_headcrabs() //Its its own proc, so they can be called by admins.
+	else
+		if(homesick >= 5)
+			mode_switch(MODE_RETURNING, list('sound/creatures/gonarch_return.ogg'))
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/gonarch_return()
+	speed = -1
+	var/turf/olddist = get_dist(src, home_nest_turf)
+	step_towards(src, home_nest_turf, speed)
+	target = home_nest_turf
+	if((get_dist(src, target)) >= (olddist))
+		frustration++
+		if(frustration > frustration_limit)
+			DestroyPathToTarget()
+			homesick++
+	else
+		frustration = 0
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/gonarch_newhome()
+	var/list/turfs = new/list()
+	for(var/turf/T in range(src, gonarch_nestfind_range))
+		if(T in range(src, 4))
+			continue
+		if(istype(T, /turf/space))
+			continue
+		if(T.density)
+			continue
+		var/lightingcount = T.get_lumcount(0.5) * 10
+		if(lightingcount > 1 || gonarch_finding_home_attempts >= gonarch_finding_home_limit_attempts)
+			continue
+		turfs += T
+	if(!turfs.len)
+		gonarch_nestfind_range += 5
+		gonarch_finding_home_attempts++
+		return
+	home_nest_turf = pick(turfs)
+	home_nest_area = get_area(home_nest_turf)
+	mode_switch(MODE_RETURNING, list('sound/creatures/gonarch_return.ogg'))
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/gonarch_rampage()
+	//We want to lose all targets, and get the hell out of here. Vision/care is restored when back in base.
+	aggro_vision_range = 0
+	vision_range = 0
+	LoseTarget()
+	homesick = 0
+	for(var/mob/living/carbon/C in hearers(gonarch_screech_range, src))
+		to_chat(C, "<span class='warning'><font size='3'><b>You hear a ear piercing shriek and your senses dull!</font></b></span>")
+		C.Weaken(25)
+		C.MinimumDeafTicks(20)
+		C.Stuttering(20)
+		C.Stun(25)
+		C.Jitter(150)
+	var/i //Workaround to avoid compiler complaints and use faster for-loops
+	for(i in 1 to 4)
+		spawn_headcrabs(FALSE)
+	for(var/obj/structure/window/W in view(gonarch_screech_range))
+		W.deconstruct(FALSE)
+	mode_switch(MODE_NEWHOME, 'sound/creatures/gonarch_initialise.ogg')
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/spawn_headcrabs(spawn_as_children = TRUE)
+	var/list/spawn_types = list(/mob/living/simple_animal/hostile/headcrab, /mob/living/simple_animal/hostile/headcrab/fast, /mob/living/simple_animal/hostile/headcrab/poison)
+	var/type_to_spawn = pick(spawn_types)
+	var/mob/living/simple_animal/hostile/headcrab/B = new type_to_spawn(get_turf(src))
+	var/list/birth_sounds = list('sound/creatures/gonarch_birth1.ogg', 'sound/creatures/gonarch_birth2.ogg', 'sound/creatures/gonarch_birth3.ogg')
+	var/chosen_sound = pick(birth_sounds)
+	playsound(get_turf(src), chosen_sound, 200, 1, 8)
```

Comment:
You can just do `playsound(src` because the proc gets the turf under the hood, same for other occurences of this. Additionally the 4th argument (`vary`) is a boolean so you can use `TRUE`